index.js

```diff
@@ -41,13 +41,17 @@ module.exports = function (cb) {
 			isReachable(hostnames, cb);
 		}
 
-		udpSocket.unref();
+		udpSocket.close();
+		udpSocket = null;
 	});
 
 	udpSocket.send(payload, 0, payload.length, 53, server, function () {
 		setTimeout(function () {
 			// We ran into the timeout, we're offline with high confidence
 			cb(null, false);
+			if (udpSocket) {
+				udpSocket.close();
+			}
 		}, timeout);
 
 		udpSocket.unref();
```

Comment:
I added these `unref`s, but I think it doesn't really make sense if you think about it: We definitely don't want to close a programm during a active online-check.